### PR TITLE
20230505 replication groundwork

### PR DESCRIPTION
**Pull request by Firstyear**
This is more work on replication - this cleans up some of the RUV maintenance code so that it's consistent after the initial content refresh. To achieve this some changes to memberof and dyngroups were needed. However now that the two RUV's on the replicas are in sync, we can now start to develop incremental replication! 

- [ x ] cargo fmt has been run
- [ x ] cargo clippy has been run
- [ x ] cargo test has been run and passes
- [ ] book chapter included (if relevant)
- [ ] design document included (if relevant)

### CODE REVIEW

File: server/lib/src/entry.rs [hunk 1]
```diff
@@ -2401,7 +2403,15 @@ where
             .eclog
             .add_ava_iter(&self.valid.cid, attr, std::iter::once(value.clone()));
         */
-        self.add_ava_int(attr, value)
+        self.add_ava_int(attr, value);
+    }
+
+    pub fn add_ava_if_not_exist(&mut self, attr: &str, value: Value) {
+        // This returns true if the value WAS changed!
```
Comment (yaleman) [inline server/lib/src/entry.rs:2410]: can we add/move this return value docstring to the ada_ava_int function? 🤔 going by the implementation it returns false if it didn't exist before, but was added which is surprising

**Review by yaleman — APPROVED** [review]

**Comment by yaleman** [comment]
Ruh roh, tests be failin'

File: server/lib/src/entry.rs (hunk 1 above)
Comment (Firstyear) [reply server/lib/src/entry.rs:2410]: It's the same as the official rust btreeset/hashset, that's why it follows the same pattern. Yes we can move the doc string. :)

**Comment by Firstyear** [comment]
> Ruh roh, tests be failin'

The failing line of code doesn't exist and is pointing at a test case that doesn't exist? What the fiuck?

**Pull request merged**